types/react-dual-listbox/index.d.ts

```diff
@@ -0,0 +1,334 @@
+// Type definitions for react-dual-listbox 2.2
+// Project: https://github.com/jakezatecky/react-dual-listbox
+// Definitions by: Justin Hall <https://github.com/wKovacs64>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.9
+
+import * as React from 'react';
+
+/** A value-based option. */
+export interface ValueOption<T> {
+    /**
+     * Whether the option is disabled or not.
+     *
+     * @default false
+     */
+    disabled?: boolean;
+    /** Adds the HTML `title` attribute to the option. */
+    title?: string;
+    /** The option label. */
+    label: string;
+    /** The option value. */
+    value: T;
+}
+
+/** A category with other categories and values. */
+export interface CategoryOption<T> {
+    /**
+     * Whether the category is disabled or not.
+     *
+     * @default false
+     */
+    disabled?: boolean;
+    /** Adds the HTML `title` attribute to the option. */
+    title?: string;
+    /** The category label. */
+    label: string;
+    /** The category child options. */
+    options: Array<Option<T>>;
+}
+
+/** Valid options include values and categories. */
+export type Option<T> = ValueOption<T> | CategoryOption<T>;
+
+/** A filter. */
+export interface Filter<T> {
+    /** Available options. */
+    available: T[];
+    /** Selected options. */
+    selected: T[];
+}
+
+/** Properties common to every `DualListBox`. */
+export interface CommonProperties<T> {
+    /**
+     * Available options.
+     *
+     * @example
+     * const options = [
+     *   { value: 'one', label: 'One'},
+     *   { value: 'two', label: 'Two'},
+     * ];
+     * <DualListBox options={options} />
+     */
+    options: Array<Option<T>>;
+    /**
+     * Selected options.
+     *
+     * @example
+     * <DualListBox options={options} selected={['one']} />
+     */
+    selected?: T[];
+    /**
+     * A React function `ref` to the "selected" list box.
+     */
+    selectedRef?: ((availableInput: HTMLSelectElement) => void) | null;
+    /**
+     * Override the default alignment of action buttons.
+     *
+     * @default "middle"
+     *
+     * @example
+     * <DualListBox options={options} alignActions="top" />
+     */
+    alignActions?: 'top' | 'middle';
+    /**
+     * If true, duplicate options will be allowed in the selected list box.
+     *
+     * @default false
+     *
+     * @example
+     * <DualListBox options={options} allowDuplicates />
+     */
+    allowDuplicates?: boolean;
+    /**
+     * A subset of the `options` array to optionally filter the available list box.
+     *
+     * @example
```

Comment:
please remove examples. I think all users know how they have to use component properties.

Reply:
Thank you, agreed - that was left over from the original effort and I didn't even think about it. Good call.